### PR TITLE
SEQ Report Action/Comment Updates

### PR DESCRIPTION
**Pull request by NicholasMoser**
This PR provides a few enhancements:

- Multiple comments can be left on the same line now
- Iruka has a few new comments added to help understand chr control flow
- Provide descriptions for most actions in the action table

### CODE REVIEW

File: src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java [hunk 1]
```diff
@@ -25,178 +29,190 @@ public class Comments {
...
+    iru0000Comments.put(0xAA8, "// Action Offset Lookup Branch #1");
+
+    iru0000Comments.put(0xB08, "// Action Offset Lookup Branch #2 (Happy path)");
+    iru0000Comments.put(0xB18, "// Load current action id into gpr2");
+    iru0000Comments.put(0xB20, "// Convert action id to table offset");
+    iru0000Comments.put(0xB28, "// Load memory address of chr_tbl into gpr19");
+    iru0000Comments.put(0xB30, "// Loads action offset from chr_tbl");
+
+    iru0000Comments.put(0x9B88, "// Set chr_tbl address to *seq_p_sp->field_0x260 + offset 0x230");
+    iru0000Comments.put(0x9E7C, "// Set chr_tbl address to *seq_p_sp->field_0x260 + offset 0x230");
```
Comment (Athkore) [inline src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java:100]: These should be *seq_p_sp->field_0x38 + offset 0x230

File: src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java [hunk 2]
```diff
@@ -25,178 +29,190 @@ public class Comments {
...
   }
 
-  private static Map<String, Map<Integer, String>> getAllComments() {
-    Map<String, Map<Integer, String>> allComments = new HashMap<>();
+  private static Map<String, Multimap<Integer, String>> getAllComments() {
+    Map<String, Multimap<Integer, String>> allComments = new HashMap<>();
 
-    Map<Integer, String> ank0000Comments = new HashMap<>();
+    Multimap<Integer, String> ank0000Comments = ArrayListMultimap.create();
     ank0000Comments.put(0x5D50, "// Scale chr model larger by 1.1x");
```
Comment (Athkore) [inline src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java:48]: Is the function to scale the chr model the full 
```
130B203F 00000003 0303003F 3F8CCCCD 0303003F 
3F8CCCCD 0303003F 3F8CCCCD 0B0A006C 00000000 
090A006C 00000000 0C095300 00000074 53000000
00000074 040E5300 00000000 3F000000 000001E0 01450000
```
or is it just the 130B203F?
In the latter case, wouldn't it be better to add the info to the opcode instead?

File: src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java [hunk 3]
```diff
@@ -611,721 +612,315 @@ public class Seq {
...
+    actionDescriptions.put(0x09D, "Weird land");
+    actionDescriptions.put(0x09E, "Weird land");
+    actionDescriptions.put(0x09F, "Weird land");
+    actionDescriptions.put(0x0A0, "5B");
+    actionDescriptions.put(0x0A1, "6B");
+    actionDescriptions.put(0x0A2, "4B");
+    actionDescriptions.put(0x0A3, "2B");
+    actionDescriptions.put(0x0A4, "DB (Duplicate)");
+    actionDescriptions.put(0x0A5, "DB");
+    // Actions between the above and below are unused
```
Comment (Athkore) [inline src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java:782]: These comments look kinda random

File: src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java [hunk 4]
```diff
@@ -611,721 +612,315 @@ public class Seq {
...
+    actionDescriptions.put(0x1D1, "Activated 3-Man X #2");
+    actionDescriptions.put(0x1D2, "Activated 3-Man X #3");
+    actionDescriptions.put(0x1D3, "Activated 3-Man X #4");
+
+    actionDescriptions.put(0x1D5, "Activated 3-Man X #5");
+
+    actionDescriptions.put(0x1DF, "Activated 3-Man X #6");
+    actionDescriptions.put(0x1E0, "Getting thrown (front)");
+    actionDescriptions.put(0x1E1, "Getting thrown (back)");
+    actionDescriptions.put(0x1E2, "Unknown");
```
Comment (Athkore) [inline src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java:913]: I think this one should be "Getting thrown (air)", even if the act id listed in game is 11E2 and not 1E2

Comment (Athkore) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java:913]: Pretty sure about this, getting hit by super is listed as 11E3 as well as getting thrown is listed as 11E0 and 11E1 respectively

**Review by Athkore — COMMENTED** [review]
Added some comments throughout :+1:

File: src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java (hunk 2 above)
Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java:48]: It's the full function. Might be good to add some kind of comment syntax here to indicate it is function documentation. Let me see how assembly code usually documents functions...

Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java:48]: `;` seems pretty popular for general comments. I've also seen `#` and `/* */`. Maybe it's worth just adding a function name to make it clear?

Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java:49]: Maybe something like this?
image
https://github.com/NicholasMoser/GNTool/pull/97/commits/09a5c5ec1a473b2c9bce9e294858bb7907e49148

File: src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java (hunk 3 above)
Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java]: Fair point, removed https://github.com/NicholasMoser/GNTool/pull/97/commits/f181e59401c8f482c71cee65c1cf0cfcb6293b6e

File: src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java (hunk 1 above)
Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/comment/Comments.java]: Good catch, fixed in https://github.com/NicholasMoser/GNTool/pull/97/commits/aeff254e1fa5072e38b252ecb7ae6d99946bb0dc

File: src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java (hunk 4 above)
Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java:913]: > even if the act id listed in game is 11E2 and not 1E2

I've noticed this as well. I believe that all actions are forced to 12 bits (0xFFF) and the higher bits represent some kind of flags. I feel like I've seen some SEQ code showing this, let me see if I can find it.

Comment (NicholasMoser) [reply src/main/java/com/github/nicholasmoser/gnt4/seq/Seq.java]: Just did some testing and I can confirm that you are correct, nice!
Updated here: https://github.com/NicholasMoser/GNTool/pull/97/commits/06b9f521a07267e5054e5d9aeaccf2027e877018

**Comment by Athkore** [comment]
Looks good to me now :+1:

**Pull request merged**